### DISCUSSION TITLE
Limits on number of policies for evaluating

### DISCUSSION THREAD

**Discussion by ash663** [open-policy-agent/community#467] [OPA and Rego]
Hi,
 
I just started looking into OPA, and was curious to know if there are any limits on number of policies in a Bundle, and how that correlates with performance for authorization decisions (in terms of end-to-end latency from when a client calls the OPA server for a decision, and receives a decision)?

My use-case could potentially grow to millions of policies, so if anyone in the community has already performed some tests on the limits, I'd appreciate any data that you could provide.

Thanks!

**Comment by ashutosh-narkar** [comment]
Hey @ash663, there is no limit on number of policies in a bundle. The default size of a bundle is 1 GB which is configurable. Wrt policy performance check out [this](https://www.openpolicyagent.org/docs/latest/policy-performance/) section on how that can be improved. There are ways to benchmark and profile your policies to help with understanding any performance issues.
